caikit/runtime/service_generation/protoable.py

```diff
@@ -91,6 +99,30 @@ def handle_protoables_in_union(arg_type: Type) -> Type:
     log.debug("Skipping non-protoable argument type [%s]", arg_type)
 
 
+def get_union_list_type(field_name: str, union_protoables: List) -> Type[DataBase]:
+    """Create a union from list type objects"""
+    common_dm_package = caikit.interfaces.common.data_model
+    param_list = []
+    for arg in union_protoables:
+        if get_origin(arg) == list:
+            arg_type = get_args(arg)[0]
```

Comment:
I think for pedanticism, we need to validate that `get_args(arg)` is not empty (e.g. `foo: Union[list, int]`) unless we somehow know that this will be filtered out by `is_protoable_type`. If we do know that that code path is impossible, it's worth a `NOTE` comment here to avoid peoples' eyes catching on a potentially dangerous index operation.